CHANGELOG.md

```diff
@@ -1,5 +1,359 @@
+  [pull/257]: https://github.com/cloudflare/wrangler/pull/257
+
+  - **Make it more clear that you can install Wrangler though npm - [zackbloom], [pull/241]**
+
+  [zackbloom]: https://github.com/zackbloom
+  [pull/241]: https://github.com/cloudflare/wrangler/pull/241
+
+  - **Document (lightly) the Wrangler 1.0.0 release - [signalnerve], [pull/204]**
+
+  [signalnerve]: https://github.com/signalnerve
+  [pull/204]: https://github.com/cloudflare/wrangler/pull/204
+
+  - **Pkg readme - [ashleygwilliams], [pull/196]**
+
+  [ashleygwilliams]: https://github.com/ashleygwilliams
+  [pull/196]: https://github.com/cloudflare/wrangler/pull/196
+
+- ### 🔧 Maintenance
+
+  - **Better error printing - [granjef3], [pull/327]**
+
+  We've updated how we log errors in Wrangler's output to make it a little easier to read. If you're a Rust developer interested in _how_ we did this, check out the [pull request][pull/327]!
+
+  [granjef3]: https://github.com/granjef3
+  [pull/327]: https://github.com/cloudflare/wrangler/pull/327
+
+  - **Always use multipart upload - [ashleymichal], [issue/280][pull/328]**
+
+  We've updated how Wrangler uploads projects to the Workers runtime - as Wrangler supports the entire Workers ecosystem (including tools like [Workers KV][kv], the publishing process should use [multipart forms](https://dev.to/sidthesloth92/understanding-html-form-encoding-url-encoded-and-multipart-forms-3lpa) to allow uploading all of the data that represents a Workers project in a single step.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [issue/280]: https://github.com/cloudflare/wrangler/issue/280
+  [pull/328]: https://github.com/cloudflare/wrangler/pull/328
+
+  - **feat: unify upload form building - [ashleymichal], [pull/329]**
+
+  This PR brings consistency to the way that metadata is handled during Wrangler's `preview` and `build` commands: previously, a `metadata.json` file handled the bindings for your Wrangler project, but because it was handled differently for various project types (such as `webpack` or `rust`), it led to inconsistent behavior during the upload process. This PR removes usage of `metadata.json` in favor of building metadata for each project type in-memory, which will then be uploaded directly to Workers platform. This work is foundational for improved Workers KV support in Wrangler 🙌
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [pull/329]: https://github.com/cloudflare/wrangler/pull/329
+
+  - **Adds integration tests for wrangler preview - [EverlastingBugstopper], [pull/363]**
+
+  Wrangler now includes integration tests for `wrangler preview`, testing every project type that Wrangler supports with our [preview service](https://cloudflareworkers.com/).
+
+  [everlastingbugstopper]: https://github.com/EverlastingBugstopper
+  [pull/363]: https://github.com/cloudflare/wrangler/pull/363
+
+  - **Add issue templates for bug reports and feature requests - [gabbifish], [issue/250][pull/350]**
```

Comment:
i'd move this to feature actually since it's user facing! (i bet i mislabelled it i'm the worst, sorry)